src/server/accounting.c

```diff
@@ -353,244 +390,138 @@ get_walltime(job *jp, int res)
  * @par	Functionality:
  *	Used by account_jobstr() and account_jobend()
  *
- *
  * @param[in]	pjob	- pointer to job structure
- * @param[in]	type	- account record type
  * @param[in]	buf	- buffer holding the data that will be stored in
  *			  accounting logs.
- * @param[in]	len	- number of characters in 'buf' still available to
- *			  store data.
- * @return	char *
- * @retval	pointer to 'buf' containing new data.
+ * @param[in]	len	- pointer to the size of 'buf'.
+ * @return	void
  *
  */
-static char *
-acct_job(job *pjob, int type, char *buf, int len)
+static void
+acct_job(job *pjob, int type, char **buf, int *len)
 {
-	pbs_list_head attrlist;
-	int	  i, k;
-	int       nd;
-	svrattrl *pal;
-	char *pb;
-	int	att_index;
-	int	len_orig;
-	char	save_char;
-	int	old_perm;
+	pbs_list_head 	attrlist;
+	svrattrl 	*pal = NULL;
+#ifdef WIN32
+	char		*win_str = NULL;
+#endif
+	char 		timestr[ACCTBUF_TIMES_NEED] = {0};
+	char            save_char = 0;
+	int		att_index;
+	int		k = 0;
+	int		old_perm;
+	int		len_orig;
 
-	pb = buf;
 	CLEAR_HEAD(attrlist);
 
 	/* gridname */
 	if (pjob->ji_wattr[(int)JOB_ATR_gridname].at_flags & ATR_VFLAG_SET) {
-		nd = strlen(pjob->ji_wattr[(int)JOB_ATR_gridname].at_val.at_str)
-			+ sizeof(GRIDNAME_FMT);
-		if (nd > len)
-			if (grow_acct_buf(&pb, &len, nd) == -1)
-				return (pb);
-
-		(void)snprintf(pb, len, GRIDNAME_FMT,
-			pjob->ji_wattr[(int)JOB_ATR_gridname].at_val.at_str);
-		i = strlen(pb);
-		pb  += i;
-		len -= i;
+		if (!pbs_strcat(buf, len, GRIDNAME_FMT)) 
+			return;
+		if (!pbs_strcat(buf, len, pjob->ji_wattr[(int)JOB_ATR_gridname].at_val.at_str))
+			return;
+		if (!pbs_strcat(buf, len, " "))
+			return;
 	}
 
 	/* user */
-	nd = sizeof(USER_FMT) + strlen(pjob->ji_wattr[(int)JOB_ATR_euser].at_val.at_str);
-	if (nd > len)
-		if (grow_acct_buf(&pb, &len, nd) == -1)
-			return (pb);
+	if (!pbs_strcat(buf, len, USER_FMT)) 
+		return;
 #ifdef WIN32
-	(void)snprintf(pb, len, USER_FMT,
-		replace_space(pjob->ji_wattr[(int)JOB_ATR_euser].at_val.at_str,
-		acctlog_spacechar));
+	win_str = replace_space(pjob->ji_wattr[(int)JOB_ATR_euser].at_val.at_str, acctlog_spacechar);
```

Comment:
It will be then consistent for all the attributes even if it is not needed for some. Here we are replacing the spaces only in the case of windows and only attributes specific to that will be processed.